exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -241,27 +214,23 @@ private void checkTransactions(List<TransactionMessage> transactionMessages) {
   }
 
   private void checkTransaction(TransactionMessage transactionMessage) {
-    short serviceId = transactionMessage.getServiceId();
+    Integer serviceId = transactionMessage.getServiceId();
     // As transactions of time service might be submitted in TestKit that has that service
     // activated, those transactions should be considered valid, even though time service is not
     // contained in 'services'
-    if (serviceId == TIME_SERVICE_ID) {
+    if (serviceId.equals(timeServiceId)) {
       return;
     }
-    if (!services.containsKey(serviceId)) {
-      String message = String.format("Unknown service id (%s) in transaction (%s)",
-          serviceId, transactionMessage);
-      throw new IllegalArgumentException(message);
-    }
-    Service service = services.get(serviceId);
-    RawTransaction rawTransaction = RawTransaction.fromMessage(transactionMessage);
+    ServiceWrapper serviceWrapper = getServiceRuntime().getServiceById(serviceId);
+    TransactionConverter txConverter = serviceWrapper.getTxConverter();
```

Comment:
See the discussion in ServiceWrapper on whether we shall use bits of service implementation (TxConverter) outside the runtime.